features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/GatewayEnvironments/AddEditGWEnvironment.jsx

```diff
@@ -58,48 +64,73 @@ function reducer(state, { field, value }) {
 }
 
 /**
- * Render a pop-up dialog to add/edit an Gateway label
+ * Render a pop-up dialog to add/edit a Gateway Environment
  * @param {JSON} props .
  * @returns {JSX}.
  */
-function AddEditMGLabel(props) {
+function AddEditGWEnvironment(props) {
     const {
         updateList, dataRow, icon, triggerButtonText, title,
     } = props;
     const classes = useStyles();
 
+    const defaultVhost = {
+        host: '', httpContext: '', httpsPort: 443, httpPort: 80, wssPort: 8099, wsPort: 9099, isNew: true,
+    };
     const [initialState, setInitialState] = useState({
```

Comment:
IMO it's better to switch to a [`useReducer`](https://reactjs.org/docs/hooks-reference.html#usereducer) hook now, Because this state is relatively complex now , not just holding a simple element

Reply:
will be fixed in https://github.com/wso2/product-apim/issues/10003

Reply:
`useReduce` has already used in here https://github.com/wso2/carbon-apimgt/pull/9890/files#diff-0ed6d7df94aad9eb68af75d9dfce90c5eca6e3fecd548b70c71fe6f0847eefb0R88